doc/api_reference.md

```diff
@@ -12,14 +12,16 @@
   * `just_error()`
   * `stop_if_requested()`
 * Sender Algorithms
-  * `transform()`
-  * `transform_done()`
+  * `then()`
```

Comment:
This PR makes me less comfortable with our naming. I'd really prefer consistency rather than have "then" stand out like this.

Reply:
So you'd prefer it to be called `upon_value`? I'm not wild about that either.

Reply:
I would prefer that so the names are at least consistent. It's not the cleanest option.